test/powershell/Language/Scripting/Generics.Tests.ps1

```diff
@@ -0,0 +1,74 @@
+﻿Describe "Generics support" -Tags "CI" {
+    #
+    # 3 types are tested, list, stack, and dictionary.
+    #
+    # list and stack are in different assemblies, and dictionary
+    # takes more than one type parameter.
+    #
+    $listType = "system.collections.generic.list"
+    $stackType = "system.collections.generic.stack"
+    $dictionaryType = "system.collections.generic.dictionary"
+
+    It 'list[Int]' {
+        $x = [system.collections.generic.list[int]]::New()
+        $x.Add(42)
+        $x.count | Should Be 1
+    }
+
+    It 'stack[Int]' {
+        $x = [system.collections.generic.stack[int]]::New()
+        $x.Push(42)
+        $x.count | Should Be 1
+    }
+
+    It 'dictionary[string, Int]' {
+        $x = [system.collections.generic.dictionary[string, int]]::New()
+        $x.foo = 42
+        $x.foo | Should Be 42
+    }
+
+    It 'list[[Int]]' {
+        $x = [system.collections.generic.list[int]]::New()
+        $x.Add(42)
+        $x.count | Should Be 1
+    }
+
+    It 'stack[[Int]]' {
+        $x = [system.collections.generic.stack[int]]::New()
+        $x.Push(42)
+        $x.count | Should Be 1
+    }
+
+    It 'dictionary[[string], [Int]]' {
+        $x = [system.collections.generic.dictionary[string, int]]::New()
+        $x.foo = 42
+        $x.foo | Should Be 42
+    }
+
+    It 'dictionaryType[dictionary[list[int],string], stack[double]]' {
+        $x = new-object "$dictionaryType[$dictionaryType[$listType[int],string], $stackType[double]]"
+        $x.gettype().fullname -like '*double*' | Should Be $true
```

Comment:
fixed

---
In reply to: [76681174](https://github.com/Powershell/Powershell/pull/2118#discussion_r76681174) [](ancestors = 76681174)